worker/src/RTC/RtpPacket.cpp

```diff
@@ -121,21 +117,8 @@ namespace RTC
 		           payloadLength + size_t{ payloadPadding },
 		  "packet's computed size does not match received size");
 
-		auto* packet = RtpPacketPool.Allocate();
-		new (packet) RtpPacket(header, headerExtension, payload, payloadLength, payloadPadding, len);
-
-		SharedPtr shared(
-		  packet,
-		  /*Deleter*/
-		  [](RtpPacket* packet)
-		  {
-			  // Call destructor manually since memory was pre-allocated upfront.
-			  packet->~RtpPacket();
-			  // Return packet into object pool for future reuse of memory allocation.
-			  RtpPacketPool.Return(packet);
-		  });
-
-		return shared;
+		return std::allocate_shared<RtpPacket, Utils::ObjectPoolAllocator<RtpPacket>>(
+		  RtpPacket::Allocator::Pool, header, headerExtension, payload, payloadLength, payloadPadding, len);
```

Comment:
I'm wonder how custom deleter would work in C++17/20 where `destroy` method was removed :thinking: